src/main.rs

```diff
@@ -14,31 +14,30 @@
 
 mod args;
 mod config;
+mod daemon;
 mod infra;
+mod log;
 mod otlp;
 mod resources;
 mod runtime;
 mod security;
 mod services;
 mod utils;
 
-use std::io;
 use std::process::exit;
 
 use anyhow::Context;
 use clap::Parser;
 
 use args::Args;
 use config::PshConfig;
+use log::log_init;
 use opentelemetry_otlp::ExportConfig;
 use runtime::PshEngineBuilder;
 use utils::check_root_privilege;
 
 fn main() -> anyhow::Result<()> {
-    tracing_subscriber::fmt()
-        .with_max_level(tracing::Level::WARN)
-        .with_writer(io::stderr)
-        .init();
+    log_init();
```

Comment:
Logging and daemonization are two different things, it would be better to be two individual commits.